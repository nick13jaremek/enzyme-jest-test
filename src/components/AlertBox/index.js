import React from 'react';
import { useDispatch } from 'react-redux';
import { createAlert } from '../../actions';

const AlertBox = () => {
  const dispatch = useDispatch();
  const clickFn = () => {
    dispatch(createAlert('New alert message'));
  };

  return (
    <div>
    <button onClick={clickFn}>Create alert message</button> 
    </div>
  );
};

export default AlertBox;
