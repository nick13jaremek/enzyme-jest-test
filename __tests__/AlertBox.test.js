import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AlertBox from '../src/components/AlertBox';

describe('AlertBox component', () => {
  let wrapper;
  let store;
  let mockStore;

  beforeAll(() => {
    const middlewares = [];
    mockStore = configureStore(middlewares);
  });

  describe('shallow', () => {
    beforeEach(() => {
      store = mockStore({});
      wrapper = shallow(
        <Provider store={store}>
        <AlertBox />
        </Provider>);
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should have mounted the component', () => {
      expect(wrapper).toBeDefined();
    });

    test('should have an AlertBox  component', () => {
      expect(wrapper.find(AlertBox)).toHaveLength(1);
    });

    test('should not render the button of the AlertBox component', () => {
      expect(wrapper.find('button')).toHaveLength(0);
    });
  });

  describe('Mount', () => {
    beforeEach(() => {
      store = mockStore({});
      wrapper = mount(
        <Provider store={store}>
          <AlertBox />
        </Provider>);
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should have a button', () => {
      expect(wrapper.find('button')).toHaveLength(1);
    });

    test('should add a new message to the store when clicking the button', () => {
      const button = wrapper.find('button').first();
      expect(button).toBeDefined();

      button.simulate('click');

      const expectedActions = [
        {
          type: 'ALERT_CREATED',
          data: 'New alert message',
        }
      ];
      const actions = store.getActions();
      expect(actions).toEqual(expectedActions);
    });
  });
});
