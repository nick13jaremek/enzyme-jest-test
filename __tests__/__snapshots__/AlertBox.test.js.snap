// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox component Mount should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AlertBox>
    <div>
      <button
        onClick={[Function]}
      >
        Create alert message
      </button>
    </div>
  </AlertBox>
</Provider>
`;

exports[`AlertBox component shallow should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <AlertBox />
</ContextProvider>
`;
